R/calcualte_age.R

```diff
@@ -21,78 +26,88 @@
 #'
 #' @examples
 #' age <- calculate_age(
-#'   data          = readRDS(system.file("extdata", "test_df.RDS",
-#'                                       package = "cleanepi")),
+#'   data = readRDS(system.file("extdata", "test_df.RDS",
+#'     package = "cleanepi"
+#'   )),
 #'   target_column = "dateOfBirth",
-#'   end_date      = Sys.Date(),
-#'   age_in        = "months",
-#'   na_strings    = "-99"
+#'   end_date = Sys.Date(),
+#'   age_in = "months",
+#'   age_remainder_unit = "days",
+#'   na_strings = "-99"
 #' )
 calculate_age <- function(data,
                           target_column = NULL,
-                          end_date      = Sys.Date(),
-                          age_in        = "years",
-                          na_strings    = cleanepi::common_na_strings) {
+                          end_date = Sys.Date(),
+                          age_in = "years",
+                          na_strings = common_na_strings,
+                          age_column_name = sprintf("age_%s", age_in),
+                          age_remainder_unit = c("days", "weeks", "months")) {
   checkmate::assert_data_frame(data, null.ok = FALSE)
-  checkmate::assert_vector(na_strings, null.ok = FALSE,
-                           any.missing = FALSE, min.len = 1L)
+  checkmate::assert_vector(na_strings,
+    null.ok = FALSE,
+    any.missing = FALSE, min.len = 1L
+  )
   checkmate::assert_choice(age_in,
-                           choices = c("years", "months", "weeks", "days"),
-                           null.ok = FALSE)
-  checkmate::assert_choice(target_column, choices = colnames(data),
-                           null.ok = TRUE)
-  end_date <- checkmate::assert_date(as.Date(end_date),
-                                     any.missing = FALSE, len = 1L,
-                                     null.ok = TRUE)
-  tmp_age <- remainder_days <- NULL
+    choices = c("years", "months", "weeks", "days"),
+    null.ok = FALSE
+  )
+  checkmate::assert_choice(target_column,
+    choices = colnames(data),
+    null.ok = TRUE
+  )
+  end_date <- checkmate::assert_date(
+    as.Date(end_date),
```

Comment:
Should we be more thorough (whatever that means) here to ensure that the user enters the right date format?  I think the `as.Date()` coercion here could go horribly wrong with bad inputs. 

For example, `checkmate::assert_date(as.Date("2-2-21"))` passes but it's actually not properly coerced.

```r
as.Date("2-2-21")
[1] "0002-02-21"
```

One option would be to improve the documentation of `end_date` to state the specific acceptable formats.

Reply:
Yes agreed - and ideally this would have some upper limit - although there may be some use case in which a future date would need to be used? I can't immediately think of one.

Reply:
Removed coercion in https://github.com/epiverse-trace/cleanepi/pull/83/commits/0c647d294f55ee3e17cb679950cdd28390da82f5

Reply:
Is the assumption here that the user/pipeline will coerce the date before passing it to this function?

Reply:
@pratikunterwegs - I think that we don't need an upper limit for the end day. If a user needs to use a future date, we have provided the correct framework for it and other scenarios too.